index.bs

```diff
@@ -435,6 +449,30 @@ They are represented by callbacks and promises in JavaScript.
 
 </div>
 
+## Device Selection ## {#programming-model-device-selection}
+
+An {{MLContext}} interface represents a global state of neural network execution. One of the important context states is the underlying execution device that manages the resources and facilitates the compilation and the eventual execution of the neural network graph. An {{MLContext}} could be created from a specific GPU device such as {{GPUDevice}} or {{WebGLRenderingContext}} that is already in use by the application, in which case the corresponding {{GPUBuffer}} or {{WebGLBuffer}} resources used as graph constants, as well as the {{GPUTexture}} and {{WebGLTexture}} as graph inputs must also be created from the same device. In a multi-adapter configuration, the device used for {{MLContext}} must be created from the same adapter as the device used to allocate the resources referenced in the graph.
+
+In a situation when a GPU context executes a graph with constants or inputs given as {{ArrayBufferView}}, the content of such constant or input is automatically uploaded from the system memory to the GPU memory. Likewise, the content of the execution result is downloaded to the system memory as {{ArrayBufferView}} upon requested.
```

Comment:
Should we info the two aspects of a CPU context when executing a graph with `ArrayBufferView`:
1. no data moving across device
2. may still have layout format conversions

Reply:
Updated. Please have a look.